### PR TITLE
fix: module prefix added twice on expand

### PR DESCRIPTION
**Pull request by fgvieira**
### Description

When using modules with prefixes, sometimes the prefix is added twice (e.g. if there are two rules with expands). For example, using the `prefix_replace`:
```
    replace_prefix:
        {"results/": "results/testmodule/"}
```

Snakemake throws the error:
```
WorkflowError:
MissingInputException: Missing input files for rule test_b:
    output: results/testmodule/b/test.out
    wildcards: name=test
    affected files:
        results/testmodule/testmodule/a/test.out
MissingInputException: Missing input files for rule test_c:
    output: results/testmodule/b/test.out
    wildcards: name=test
    affected files:
        results/testmodule/testmodule/b/test.out
```

The same happens if using the prefix:
```
    prefix:
	"test"
```
with Snakemake throwing the error:
```
WorkflowError:
MissingInputException: Missing input files for rule test_c:
    output: test/results/b/test.out
    wildcards: name=test
    affected files:
        test/test/results/b/test.out
MissingInputException: Missing input files for rule test_b:
    output: test/results/b/test.out
    wildcards: name=test
    affected files:
        test/test/results/a/test.out
```

With more verbose logs, it seems that the prefix is being added before and after `expand`:
```
[...]
Path results/a/{name}.out, converted to test/results/a/{name}.out
Path test/results/a/test.out, converted to test/test/results/a/test.out
Path results/b/{name}.out, converted to test/results/b/{name}.out
Path test/results/b/test.out, converted to test/test/results/b/test.out
Path results/b/{name}.out, converted to test/results/b/{name}.out
Path test/results/b/test.out, converted to test/test/results/b/test.out
Path test/results/b/test.out, converted to test/results/b/test.out
Path did not change after prefix replace
[...]
```
I think `prefix` should not be changed after `expand`, since `prefix_replace` might replace the wildcards, so I added a function to copy the file attributes after `expand`.

### QC

* [x] The PR contains a test case for the changes or the changes are already covered by an existing test case.
* [x] The documentation (`docs/`) is updated to reflect the changes or this is not necessary (e.g. if the change does neither modify the language nor the behavior or functionalities of Snakemake).

### CODE REVIEW

File: snakemake/io.py [hunk 1]
```diff
@@ -1305,10 +1306,20 @@ def flatten(wildcard_values):
...
 
+        def copy_flags(from_path, dest_path):
+            if hasattr(from_path, "flags"):
+                dest_path = AnnotatedString(dest_path)
+                dest_path.flags.update(from_path.flags)
+                if is_flagged(dest_path, "multiext"):
+                    dest_path.flags["multiext"] = self.apply_default_storage(
+                        self.replace_prefix(dest_path.flags["multiext"], property)
+                    )
+            return dest_path
```
Comment (johanneskoester) [inline snakemake/io.py:1317]: I fear that we forget the reason for this later. Can you add some comments?

Comment (johanneskoester) [reply snakemake/io.py:1317]: And why the special case for multiext?

Comment (johanneskoester) [reply snakemake/io.py:1317]: And maybe multiext should then also occur in the test case?

**Review by johanneskoester — CHANGES_REQUESTED** [review]
Thanks a lot! I think the code needs some documentation though

File: snakemake/io.py (hunk 1 above)
Comment (fgvieira) [reply snakemake/io.py:1317]: Sure, I can add some comments.
Actually I am not sure if we need that special case for `multiext`.. I'll add it to the test case and check.

**Review by johanneskoester — APPROVED** [review]

**Pull request merged**